src/io/mod.rs

```diff
@@ -0,0 +1,6 @@
+#[cfg(target_family = "unix")]
+mod unix;
+#[cfg(target_family = "unix")]
+pub(crate) use unix::File;
+#[cfg(target_family = "unix")]
+pub use unix::IoDriver;
```

Comment:
It would be good to fail compilation with descriptive error for non-supported platforms with `compile_error!` macro (https://doc.rust-lang.org/std/macro.compile_error.html). 

```rust
#[cfg(not(target_family = "unix"))]
compile_error!("Specified target platform is not supported (only unix family supported)");
```